app/controllers/vm_remote.rb

```diff
@@ -94,4 +106,36 @@ def console_after_task(console_type)
       javascript_open_window(url)
     end
   end
+
+  def native_console_launch_task
+    record = identify_record(params[:id], VmOrTemplate)
+
+    task_id = record.native_console_connection_queue(session[:userid])
+    unless task_id.kind_of?(Integer)
+      add_flash(_("Native console access failed: Task start failed"), :error)
+    end
+
+    if @flash_array
+      javascript_flash(:spinner_off => true)
+    else
+      initiate_wait_for_task(:task_id => task_id, :action => 'native_console_task_complete')
+    end
+  end
+
+  def native_console_task_complete
+    miq_task = MiqTask.find(params[:task_id])
```

Comment:
Do we really need to search for the `MiqTask` here, as far as I see, it's just "validation" which could be done in the next step when opening the window.

Reply:
Yes it's only to test if the task has results. You're right I'll try to move it in the *launch_native_console* so we can remove the first find.